packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -347,6 +350,7 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 		return {
 			sequenceNumber: this.lastSequenceNumber,
 			opsSize: this.opsSize > 0 ? this.opsSize : undefined,
+			deltaManagerState: this._disposed ? "disposed" : this._closed ? "closed" : "open",
```

Comment:
this will help differentiate closes due to dispose/close from disconnects for other reasons

Reply:
Don't the reason strings already do that? Won't this always be the same as the container state?

Reply:
no. during disposal it get weird. The first thing that happens is dispose on the dm, which disconnects, but only after dm disposed finishes does the container dispose kick in. in the logs it looks like:
<HTML><HEAD><TITLE>From Clipboard</TITLE></HEAD><BODY><!--StartFragment-->

Data_reason | Data_containerLifecycleState | Data_containerConnectionState
-- | -- | -- | -- | --
Closing DeltaManager |  loaded | Disconnected


<!--EndFragment--></BODY></HTML>